src/resolver.ts

```diff
@@ -3092,17 +3092,18 @@ export class Resolver extends DiagnosticEmitter {
             let fieldPrototype = <FieldPrototype>member;
             let fieldTypeNode = fieldPrototype.typeNode;
             let fieldType: Type | null = null;
-            // TODO: handle duplicate non-private fields specifically?
+            let existingField: Field | null = null;
+            if (base) {
+              let baseMembers = base.members;
+              if (baseMembers !== null && baseMembers.has(fieldPrototype.name)) {
+                let baseField = assert(baseMembers.get(fieldPrototype.name));
+                assert(baseField.kind == ElementKind.FIELD);
```

Comment:
Doesn't seem to be guaranteed, could as well be a method for example and then crash the compiler.

Reply:
Huh, I thought I could keep that assert in there because it was in the old code, and from my testing I wasn't able to crash the compiler pre-PR. I'll change that right away

Reply:
Makes me wonder if this is checked earlier already, when the class is initialized? Could that be?